src/opentelemetry.cr

```diff
@@ -114,12 +114,16 @@ module OpenTelemetry
   # end
   # ```
   def self.configure
-    CONFIG.service_name = ENV["OTEL_SERVICE_NAME"]?
-
     yield CONFIG
 
-    # No need to configure the resource if already assigned within yield block
-    return unless CONFIG.resource.nil?
+    if env_service_name = ENV["OTEL_SERVICE_NAME"]?
+      # OTEL_SERVICE_NAME var takes precedence over yield block configuration
+      CONFIG.service_name = env_service_name
+    elsif CONFIG.resource
+      # If resource was already configured in yield block and OTEL_SERVICE_NAME
+      # ENV var isn't available means we don't need to build the resource
+      return
+    end
```

Comment:
Remember how we were happy about avoiding the conditional? Oh how young and naive we were. 😄